src/main/java/ch/saunah/saunahbackend/controller/UserController.java

```diff
@@ -80,5 +81,51 @@ public ResponseEntity<String> resetPassword(@RequestBody ResetPasswordBody reset
         return ResponseEntity.ok("success");
     }
 
+    @Operation(description = "Returns a list of all Users.")
+    @GetMapping(path="users")
+    public @ResponseBody
+    List<UserResponse> getAllUsers() {
+        return userService.getAllUser().stream().map(x -> new UserResponse(x)).collect(Collectors.toList());
+    }
+
+    @Operation(description = "Returns the user with the ID specified.")
+    @GetMapping(path="users/{id}")
+    public @ResponseBody ResponseEntity<UserResponse> getUser(@PathVariable(value = "id", required = true) Integer id, Principal principal) throws AuthenticationException {
+        User currentUser = userService.getUserByMail(principal.getName());
+        if (currentUser.getRole().equals(UserRole.ADMIN) || currentUser.getId().equals(id)) {
+            return ResponseEntity.ok(new UserResponse(userService.getUser(id)));
+        }
+        throw new AuthenticationException("user is not authenticated to view other users than himself");
+    }
+
+    @Operation(description = "Returns the user with the ID specified.")
+    @PutMapping(path="users/userRole/{id}")
+    public @ResponseBody ResponseEntity<UserResponse> changeUserRole(@PathVariable(value = "id", required = true) Integer id,  Principal principal) throws AuthenticationException {
+        User currentUser = userService.getUserByMail(principal.getName());
+        if (currentUser.getRole().equals(UserRole.ADMIN)) {
+            User changedUser = userService.getUser(id);
+            if (changedUser.getRole().equals(UserRole.ADMIN)) {
+                return ResponseEntity.ok(new UserResponse(userService.editUserRole(changedUser, UserRole.USER)));
+            } else {
+                return ResponseEntity.ok(new UserResponse(userService.editUserRole(changedUser, UserRole.ADMIN)));
+            }
+        }
+        throw new AuthenticationException("user is not authenticated to change the userRole");
+    }
+
+    @Operation(description = "Returns the user with the ID specified.")
+    @PutMapping(path="users/{id}")
+    public @ResponseBody ResponseEntity<UserResponse> editUser(@PathVariable(value = "id", required = true) Integer id, Principal principal, @RequestBody SignUpBody signUpBody) throws AuthenticationException {
+        User currentUser = userService.getUserByMail(principal.getName());
+        if (currentUser.getRole().equals(UserRole.ADMIN) || currentUser.getId().equals(id)) {
+            return ResponseEntity.ok(new UserResponse(userService.editUser(id, signUpBody)));
```

Comment:
This returns an object of type `UserResponse`. `UserResponse`, however, contains some sensitive information, which I think should not be returned for such an API call, such as `passwordHash` or `resetpasswordhash`.

Reply:
password Hash and resetpasswordHash have been removed from User Response, therefore it no longer should contain senstive data